src/__test__/services/holidayRecommendationService.test.js

```diff
@@ -2,16 +2,48 @@ import { getHolidayRecommendations } from '../../services/holidayRecommendationS
 
 describe('holidayService', () => {
   describe('getHolidayRecommendations', () => {
-    it('returns all holiday recommendations in given year', () => {
+    it('returns all holiday recommendations in given year when queried at the beginning of the year', () => {
+      jest
+        .spyOn(global.Date, 'now')
+        .mockImplementation(() =>
+          new Date('2020-01-01T00:00:00.000Z').valueOf()
+        );
+
       const holidayRecommendations = getHolidayRecommendations(2020, 8);
       expect(holidayRecommendations.length).toEqual(10);
     });
 
+    it('returns only the recommendations with start date that is equal to or later than current date', () => {
+      jest
+        .spyOn(global.Date, 'now')
+        .mockImplementation(() =>
+          new Date('2020-05-27T00:00:00.000Z').valueOf()
+        );
+
+      const holidayRecommendations = getHolidayRecommendations(2020, 8);
+
+      expect(
+        holidayRecommendations.every(
+          (recommendation) => recommendation.startDate >= Date.now()
+        )
+      ).toBe(true);
+
+      expect(holidayRecommendations.length).toEqual(5);
+    });
+
     it('returns holiday recommendations sorted by numberOfAnnualLeaves asc', () => {
       const holidayRecommendations = getHolidayRecommendations(2020, 5);
+
       expect(
-        holidayRecommendations[0].numberOfAnnualLeaves <=
-          holidayRecommendations[1].numberOfAnnualLeaves
+        holidayRecommendations.every((recommendation, index) => {
+          const nextRecommendation = holidayRecommendations[index + 1];
+
+          return (
+            nextRecommendation === undefined ||
```

Comment:
Good work on checking every element of the recommendations, but this part is not very clear imo. 

Would it be better to check the index before getting `nextRecommendation`? 
```suggestion
            if (index === holidayRecommendations.length) {
                return true;
            }
            
            const nextRecommendation = holidayRecommendations[index + 1];
            return recommendation.numberOfAnnualLeaves <= nextRecommendation.numberOfAnnualLeaves;
```

Reply:
Since index will always be less than the length of the list, the initial check in the suggestion will always be false

Beside that one off error which is not really the point you were trying to make, I think what you are alluding to is something along the line of
```javascript
const isLastElement = (index === holidayRecommendations.length -1);

if (isLastElement) {
    return true;
}

const nextRecommendation = holidayRecommendations[index + 1];

return recommendation.numberOfAnnualLeaves <= nextRecommendation.numberOfAnnualLeaves;
```

I think this is mostly preferential, but the declarative nature of functions like `every()` I think is more readable and more in grain with the functional programming style when written in the form of 'Get next and check such and such' rather than 'Check if the current is last, if not check..' form.